### PR TITLE
feat(udp): migrate udp domain to use sdk

### PR DESCRIPTION
**Pull request by aron-bs**
# Pull Request

## Description

- Refactor UDP domain config and handlers to use SDK
- Rename domain handlers to match updated domain config
- Remove redundant/stale files
- Refactor test suites to verify SDK domain config/handlers
- Remove redundant workspace dependencies 
- Refactor DVLA routes to use SDK

**Related Issue:** https://gdsgovukagents.atlassian.net/browse/FLEX-229

## Type of Change

Please check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (code change that does not add functionality or fix a bug)
- [x] Code cleanup (formatting, renaming)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Other (please explain)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so others can reproduce.

- [x] Unit tests
- [x] Integration tests
- [x] Manual testing (include instructions below)

## Checklist

- [x] I have run the pre-commit
- [x] I have run all necessary tests
- [x] I have updated/added all necessary tests
- [x] I have added or updated documentation
- [x] I have performed a self-review of my code
- [x] I have commented my code in hard-to-understand areas
- [x] I have refactored my code to be more readable, maintainable and removed duplications.
- [x] I have added guards around invalid inputs and edge cases such as nulls and undefined

### CODE REVIEW

File: domains/udp/src/tests/setup.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { afterAll, afterEach, beforeAll, beforeEach, expect, vi } from "vitest";
+
+vi.mock("@middy/secrets-manager", () => ({
+  default: () => ({ before: vi.fn() }),
+  secret: vi.fn((v: string) => v),
+}));
+
+// vi.mock("@middy/ssm", () => ({
+//   default: () => ({ before: vi.fn() }),
+// }));
```
Comment (pazsaragi) [inline domains/udp/src/tests/setup.ts]:
```suggestion
```

**Review by pazsaragi — DISMISSED** [review]

**Review by nathanberry97 — APPROVED** [review]
